### PR TITLE
docs: added extend documentation pages

### PR DESCRIPTION
**Pull request by shahednasser**
- Added extend entity page
- Added extend repository page
- Added extend service page
- Added details about service lifetime in the Create Service documentation.
- Updated Middlewares documentation to include details about registering resources
- Added an example documentation that illustrates how to register the logged in user
- Other small changes across pages.

Closes DOCS-661, DOCS-663, and DOCS-641

### CODE REVIEW

File: docs/content/development/fundamentals/dependency-injection.md [hunk 1]
```diff
@@ -193,6 +229,11 @@ Every fulfillment provider is registered under two names:
 - Its camel-case name. For example, the `WebshipperFulfillmentService` is registered as `webshipperFulfillmentService`.
 - `fp_` followed by its identifier. For example, the `WebshipperFulfillmentService` is registered as `fp_webshipper`.
 
+</td>
+<td>
+
+By default, it's `TRANSIENT` unless defined differently within the fulfillemnt provider service.
```
Comment (adrien2p) [inline docs/content/development/fundamentals/dependency-injection.md:235]: todo: singleton

File: docs/content/development/fundamentals/dependency-injection.md [hunk 2]
```diff
@@ -250,6 +301,11 @@ An array of all notification providers that extend the `AbstractNotificationServ
 
 `notificationProviders`
 
+</td>
+<td>
+
+\-
```
Comment (adrien2p) [inline docs/content/development/fundamentals/dependency-injection.md:307]: todo: trensient

**Review by adrien2p — CHANGES_REQUESTED** [review]

File: docs/content/development/endpoints/add-middleware.md [hunk 3]
```diff
@@ -169,7 +144,7 @@ class CustomService extends TransactionBaseService {
 export default CustomService
 ```
 
-You can learn more about lifetime in the [Create Service documentation](../services/create-service.md#service-life-time).
+Notice that the `LIFE_TIME` property has been set to `Lifetime.SCOPED`. If you want to access new registrations in the dependency container within a service, you must set the lifetime of the service either to `Lifetime.SCOPED` or `Lifetime.TRANSIENT`.
```
Comment (adrien2p) [inline docs/content/development/endpoints/add-middleware.md:147]: todo: only valid for core services since the custom services are transient by default

**Review by adrien2p — CHANGES_REQUESTED** [review]

File: docs/content/development/fundamentals/dependency-injection.md [hunk 4]
```diff
@@ -331,6 +402,11 @@ An array of every tax provider that extends the `AbstractTaxService` class.
 
 `taxProviders`
 
+</td>
+<td>
+
+\-
```
[the remark this answers is not in the document]
Comment (shahednasser) [reply docs/content/development/fundamentals/dependency-injection.md:408]: I think you mean that each item in the array is transient right? I was thinking of highlighting that too. I'll add it in other places as well

File: docs/content/development/fundamentals/dependency-injection.md [hunk 5]
```diff
@@ -232,6 +278,11 @@ Every notification provider is registered under two names:
 - Its camel-case name. For example, the `SendGridService` is registered as `sendGridService`.
 - `noti_` followed by its identifier. For example, the `SendGridService` is registered as `noti_sendgrid`.
 
+</td>
+<td>
+
+By default, it's `TRANSIENT` unless defined differently within the notification provider service.
```
[the remark this answers is not in the document]
Comment (shahednasser) [reply docs/content/development/fundamentals/dependency-injection.md:284]: According to [this](https://github.com/medusajs/medusa/blob/ebd6ab12c944d517f46aede9775d5937e7726962/packages/medusa/src/loaders/plugins.ts#L416) it's Transient, or am I checking the wrong place? 😅

**Review by adrien2p — APPROVED** [review]
LGTM

**Pull request merged**